### PR TITLE
6563 display data genre

### PR DESCRIPTION
**Pull request by Hiepsisoma**
https://edu-redmine.framgia.vn/issues/6563

### CODE REVIEW

**Comment by Hiepsisoma** [comment]
ready

File: app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java [hunk 1]
```diff
@@ -4,20 +4,42 @@
...
+import vn.framgia.phamthehung.soundcloud.data.model.Track;
+import vn.framgia.phamthehung.soundcloud.data.repository.TrackRepository;
+import vn.framgia.phamthehung.soundcloud.data.source.remote.TrackRemoteDataSource;
 
-public class GenreActivity extends AppCompatActivity {
+public class GenreActivity extends AppCompatActivity implements GenreContract.View,
+        View.OnClickListener, GenreAdapter.OnItemClickListener {
     public static final String GENRE_KEY = "GENRE_KEY";
-
+    private ImageView mImageView;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java]:
```suggestion
    private ImageView mImageGenre;
```

File: app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java [hunk 2]
```diff
@@ -4,20 +4,42 @@
...
+    private TextView mTextView;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java]:
```suggestion
    private TextView mTextShuffle;
    hoặc gì đó cho có ý nghĩa
```

File: app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java [hunk 3]
```diff
@@ -4,20 +4,42 @@
...
+    private Genre mGenre;
+    private GenrePresenter mGenrePresenter;
+    private ProgressBar mProgressBar;
+    private RecyclerView mRecyclerView;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java]:
```suggestion
    private RecyclerView mRecyclerTracks;
```

File: app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java [hunk 4]
```diff
@@ -47,4 +82,41 @@ public void initActionBar() {
...
+        mTracks = track;
+        mProgressBar.setVisibility(View.GONE);
+        mRecyclerView.setVisibility(View.VISIBLE);
+        mGenreAdapter = new GenreAdapter(this, mTracks,this);
+        mRecyclerView.setAdapter(mGenreAdapter);
+        mGenreAdapter.notifyDataSetChanged();
+    }
+
+    @Override
+    public void loadTrackFail(String message) {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreActivity.java]:
```suggestion
    public void loadTrackFailure(String message) {
```

File: app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreContract.java [hunk 5]
```diff
@@ -1,4 +1,17 @@
...
 
+import java.util.List;
+
+import vn.framgia.phamthehung.soundcloud.data.model.Genre;
+import vn.framgia.phamthehung.soundcloud.data.model.Track;
+
 public class GenreContract {
+    interface View {
+        void loadTracksSuccess(List<Track> track);
+        void loadTrackFail(String message);
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/vn/framgia/phamthehung/soundcloud/ui/genre/GenreContract.java]:
```suggestion
        void loadTrackFailure(String message);
```

File: app/src/main/res/layout/item_track.xml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        android:maxLines="1"
+        android:text="@string/title_all_music"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/text_count"
+        app:layout_constraintTop_toBottomOf="@+id/text_name_track"
+        app:layout_constraintVertical_bias="0.48" />
+
+    <ImageView
+        android:id="@+id/image_more"
+        android:layout_width="50dp"
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/layout/item_track.xml]: hardcode

File: gradle.properties [hunk 7]
```diff
@@ -11,5 +11,5 @@ org.gradle.jvmargs=-Xmx1536m
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
-
+API_KEY=a7Ucuq0KY8Ksn8WzBG6wj4x6pcId6BpU
```
Comment (NguyenMinhTuanB) [inline gradle.properties]: không đưa keey lên Github

**Comment by Hiepsisoma** [comment]
ready

**Pull request merged**